cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -844,6 +861,22 @@ func shouldDeleteNode(ctx *controllerContext, node *v1.Node, getInstance func(*c
 		klog.Errorf("Error retrieving instance %q: %v", node.Name, err)
 		return false, err
 	}
+
+	if inst != nil {
```

Comment:
can inst be nil if err was not nil, is this check unnecessary?

Reply:
Good point! Didn't see any possible where inst and err could be both nil... Except in one of the unit test, I have tweaked that test case and mentioned you there. Thanks for addressing this.